akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/EventWriter.scala

```diff
@@ -257,6 +335,115 @@ private[akka] object EventWriter {
               Behaviors.same
           }
 
+        def handleWrite(repr: PersistentRepr, replyTo: ActorRef[StatusReply[WriteAck]], calledAfterMaxSeqNr: Boolean)
+            : Unit = {
+          val persistenceId = repr.persistenceId
+          val sequenceNumber = repr.sequenceNr
+          val newStateForPid =
+            perPidWriteState.get(persistenceId) match {
+              case None =>
+                if (fillSequenceNumberGaps && sequenceNumber != 1L) {
+                  askMaxSeqNr(persistenceId, originalErrorDesc = None)
+                  StateForPid(waitingForReply = Map.empty, waitingForSeqNrLookup = Vector((repr, replyTo)))
+                } else {
+                  sendToJournal(1, Vector(repr))
+                  StateForPid(Map((repr.sequenceNr, (repr, replyTo))), currentTransactionId = 1)
+                }
+
+              case Some(state) =>
+                val expectedSeqNr =
+                  if (fillSequenceNumberGaps) state.nextExpectedSeqNr
+                  else repr.sequenceNr
+
+                if (state.seqNrlookupInProgress) {
+                  context.log.trace2(
+                    "Seq nr lookup in progress for persistence id [{}], adding sequence nr [{}] to pending",
+                    persistenceId,
+                    sequenceNumber)
+                  state.copy(waitingForSeqNrLookup = state.waitingForSeqNrLookup :+ ((repr, replyTo)))
+                } else if (sequenceNumber == expectedSeqNr) {
+                  if (state.idle) {
+                    sendToJournal(state.currentTransactionId + 1, Vector(repr))
+                    state.copy(
+                      waitingForReply = Map((repr.sequenceNr, (repr, replyTo))),
+                      currentTransactionId = state.currentTransactionId + 1)
+                  } else {
+                    // write in progress for pid, add write to batch and perform once current write completes
+                    if (state.waitingForWriteExceedingMaxBatchSize(settings.maxBatchSize)) {
```

Comment:
I relaxed this check in c8d8fbef84ce0596fc7482fa3d610f82a3ccd924, so that it's not counting FilteredPayload. Otherwise it could easily be exceeded by a larger gap. Do you think that is alright @johanandren ?